diff -Nur libpst-0.6.24-orig/src/libpst.h libpst-0.6.24/src/libpst.h
--- libpst-0.6.24-orig/src/libpst.h	2009-01-15 13:43:38.000000000 -0600
+++ libpst-0.6.24/src/libpst.h	2009-01-15 15:29:51.000000000 -0600
@@ -449,6 +449,7 @@
     struct pst_item_folder        *folder;          // data reffering to folder
     struct pst_item_contact       *contact;         // data reffering to contact
     struct pst_item_attach        *attach;          // linked list of attachments
+    struct pst_item_attach	  *current_attach;  // linked list of current attachments
     struct pst_item_message_store *message_store;   // data referring to the message store
     struct pst_item_extra_field   *extra_fields;    // linked list of extra headers and such
     struct pst_item_journal       *journal;         // data reffering to a journal entry
